src/natcap/invest/ndr/ndr_core.pyx

```diff
@@ -529,18 +529,24 @@ def ndr_eff_calculation(
 
                     neighbor_effective_retention = (
                         effective_retention_raster.get(ds_col, ds_row))
-                    if neighbor_effective_retention >= retention_eff_lulc:
-                        working_retention_eff += (
-                            neighbor_effective_retention) * outflow_weight
-                    else:
+
+                    # Case 1: downstream neighbor is a stream pixel
+                    if neighbor_effective_retention == 0:
```

Comment:
To be even more similar to the user's guide, this could directly check the stream raster instead: `if stream_raster.get(ds_row, ds_col) == 1`. But the way you have it is more efficient. I think it's good either way.

Reply:
Yeah, comparing `int`s here should be faster than possibly incurring a cache miss on the stream managed raster.  That being said, I think a module constant here would help clarify that we're checking for a stream, so I've added one in 13fac9f51